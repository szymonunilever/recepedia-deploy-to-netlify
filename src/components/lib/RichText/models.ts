export interface RichTextProps {
  className?: string;
  html: string;
}
