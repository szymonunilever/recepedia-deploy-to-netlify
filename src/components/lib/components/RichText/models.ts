import { UnileverLibraryComponent } from '../common/globalModels';

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface RichTextProps
  extends UnileverLibraryComponent<AppContent.RichTextContent> {}
