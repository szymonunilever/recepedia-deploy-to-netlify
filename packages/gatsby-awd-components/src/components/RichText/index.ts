export { default, RichText } from './RichText';
export * from './models';
