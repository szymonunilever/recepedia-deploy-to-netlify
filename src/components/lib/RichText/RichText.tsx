import React from 'react';
import { RichTextProps } from './models';

const RichText = ({ html, className = '' }: RichTextProps) => (
  <div className={className} dangerouslySetInnerHTML={{ __html: html }} />
);

export default RichText;
