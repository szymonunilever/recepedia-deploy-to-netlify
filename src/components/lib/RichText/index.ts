export { default as RichText } from './RichText';
export * from './models';
