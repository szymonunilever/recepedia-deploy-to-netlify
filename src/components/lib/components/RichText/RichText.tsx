import React from 'react';
import ReactMarkdown from 'react-markdown';
import { RichTextProps } from './models';

export const RichText = ({
  content: { text },
  type = 'html',
  className,
}: RichTextProps) => {
  return type === 'html' ? (
    <div
      data-componentname="richText"
      className={className}
      dangerouslySetInnerHTML={{ __html: text }}
    />
  ) : (
    <ReactMarkdown data-componentname="richText" className={className}>
      {text}
    </ReactMarkdown>
  );
};

export default RichText;
