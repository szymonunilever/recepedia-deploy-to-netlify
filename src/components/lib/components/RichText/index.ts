export { default } from './RichText';
export { RichText } from './RichText';
export * from './models';
