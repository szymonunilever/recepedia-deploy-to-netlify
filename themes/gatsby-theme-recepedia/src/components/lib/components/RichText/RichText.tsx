import React from 'react';
import ReactMarkdown from 'react-markdown';
import { RichTextProps } from './models';
import getComponentDataAttrs from '../../utils/getComponentDataAttrs';

export const RichText = ({
  content,
  content: { text },
  type = 'html',
  className,
}: RichTextProps) => {
  return type === 'html' ? (
    <div
      {...getComponentDataAttrs('richText', content)}
      className={className}
      dangerouslySetInnerHTML={{ __html: text }}
    />
  ) : (
    <ReactMarkdown
      {...getComponentDataAttrs('richText', content)}
      className={className}
    >
      {text}
    </ReactMarkdown>
  );
};

export default RichText;
