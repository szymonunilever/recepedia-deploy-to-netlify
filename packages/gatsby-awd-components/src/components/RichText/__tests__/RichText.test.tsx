import { shallow, mount, ReactWrapper } from 'enzyme';
import React from 'react';
import RichText from '../RichText';
import toJson from 'enzyme-to-json';

const content: AppContent.RichTextContent = {
  text: `
    <h1>Header 1</h1>
    <h2>Header 2</h2>
    <p>
    Paragraph
    <a href="http://test.html">Test link</a>
    </p>
  `,
};

const className = 'rich-text';

describe('RichText', () => {
  let wrapper: ReactWrapper;

  beforeEach(() => {
    wrapper = mount(<RichText content={content} className={className} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render text from props', () => {
    const testContent = shallow(
      <RichText content={content} className={className} />
    );
    expect(testContent.find('div[data-componentname="richText"]')).toBeTruthy();
    expect(testContent.find('h1')).toBeTruthy();
    expect(testContent.find('h2')).toBeTruthy();
    expect(testContent.find('p')).toBeTruthy();
    expect(testContent.find('a')).toBeTruthy();
  });
});
