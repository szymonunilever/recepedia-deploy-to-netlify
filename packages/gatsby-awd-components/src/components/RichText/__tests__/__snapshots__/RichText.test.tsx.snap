// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichText matches the snapshot 1`] = `
<Component
  className="rich-text"
  content={
    Object {
      "text": "
    <h1>Header 1</h1>
    <h2>Header 2</h2>
    <p>
    Paragraph
    <a href=\\"http://test.html\\">Test link</a>
    </p>
  ",
    }
  }
>
  <div
    className="rich-text"
    dangerouslySetInnerHTML={
      Object {
        "__html": "
    <h1>Header 1</h1>
    <h2>Header 2</h2>
    <p>
    Paragraph
    <a href=\\"http://test.html\\">Test link</a>
    </p>
  ",
      }
    }
    data-componentname="richText"
    data-componentview="richText"
  />
</Component>
`;
